main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -377,6 +393,116 @@ private void loadFeedInfo(GtfsTableContainer<GtfsFeedInfo> feedTable) {
           localDate.equals(GtfsFeedInfo.DEFAULT_FEED_END_DATE) ? "N/A" : localDate.toString();
       feedInfo.put(FEED_INFO_FEED_END_DATE, info == null ? "N/A" : displayDate);
     }
+
+    loadServiceDateRange(tripContainer, calendarTable, calendarDateTable);
+  }
+
+  /**
+   * Loads the service date range by determining the earliest start date and the latest end date for
+   * all services referenced with a trip\_id in `trips.txt`. It handles three cases: 1. When only
+   * `calendars.txt` is used. 2. When only `calendar\_dates.txt` is used. 3. When both
+   * `calendars.txt` and `calendar\_dates.txt` are used.
+   *
+   * @param tripContainer the container for `trips.txt` data
+   * @param calendarTable the container for `calendars.txt` data
+   * @param calendarDateTable the container for `calendar\_dates.txt` data
+   */
+  public void loadServiceDateRange(
+      GtfsTableContainer<GtfsTrip> tripContainer,
+      GtfsTableContainer<GtfsCalendar> calendarTable,
+      GtfsTableContainer<GtfsCalendarDate> calendarDateTable) {
+    List<GtfsTrip> trips = tripContainer.getEntities();
+
+    LocalDate earliestStartDate = null;
+    LocalDate latestEndDate = null;
+    if ((calendarDateTable == null) && (calendarTable != null)) {
+      // When only calendars.txt is used
+      List<GtfsCalendar> calendars = calendarTable.getEntities();
+      for (GtfsTrip trip : trips) {
+        String serviceId = trip.serviceId();
+        for (GtfsCalendar calendar : calendars) {
+          if (calendar.serviceId().equals(serviceId)) {
+            LocalDate startDate = calendar.startDate().getLocalDate();
+            LocalDate endDate = calendar.endDate().getLocalDate();
+
+            if (startDate != null || endDate != null) {
+              if (earliestStartDate == null || startDate.isBefore(earliestStartDate)) {
+                earliestStartDate = startDate;
+              }
+              if (latestEndDate == null || endDate.isAfter(latestEndDate)) {
+                latestEndDate = endDate;
+              }
+            }
+          }
+        }
+      }
+    } else if ((calendarDateTable != null) && (calendarTable == null)) {
+      // When only calendar_dates.txt is used
+      List<GtfsCalendarDate> calendarDates = calendarDateTable.getEntities();
+      for (GtfsTrip trip : trips) {
+        String serviceId = trip.serviceId();
+        for (GtfsCalendarDate calendarDate : calendarDates) {
+          if (calendarDate.serviceId().equals(serviceId)) {
+            LocalDate date = calendarDate.date().getLocalDate();
```

Comment:
Maybe you should check if the exception_type is addition before using it in the calculation.

Reply:
When only calendar_dates.txt is used, can we assume that the exception_type is always 1? @emmambd